pkg/sources/forward/data_forward.go

```diff
@@ -63,8 +63,11 @@ type DataForward struct {
 	vertexName     string
 	pipelineName   string
 	vertexReplica  int32
+	Watermark      dfv1.Watermark
 	// idleManager manages the idle watermark status.
-	idleManager wmb.IdleManager
+	idleManager      wmb.IdleManager
+	lastUpdateWM     time.Time
+	currentWatermark wmb.Watermark
```

Comment:
We have been adding attributes to the data forwarder and it's now taking care of too many things, especially watermark-related logics. I am wondering if we can move these attributes to the watermark publisher itself. That way we delegate all the watermark publishing logics to the publisher hence data forwarder doesn't need to know about how an idle source watermark is published.

Such delegation makes our system more modularized and potentially easier to unit test. 

Ideally, it would be great if we could have the following statement:

```
if len(readMesages) == 0 {
  isdf.srcWMPublisher.PublishIdleWatermarks()
}
```

and let srcWMPublisher take care of tracking updatedWM etc.